http-clients/src/main/java/org/triplea/http/client/web/socket/messages/MessageTypeListenerBinding.java

```diff
@@ -0,0 +1,25 @@
+package org.triplea.http.client.web.socket.messages;
+
+import com.google.errorprone.annotations.Immutable;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import lombok.Value;
+
+@Value(staticConstructor = "of")
+@Immutable
+// The function passed as second param isn't
+// guaranteed to be immutable, but we are
+// going to trust the future-devs to use
+// stateless lambdas.
+@SuppressWarnings("Immutable")
+public class MessageTypeListenerBinding<T, X> {
```

Comment:
Nit: I'm not a fan that template types need to end in 'T', in the case of more than one though, it's a lot more descriptive to spell out the types and what they are meant to represent. EG:
```
<ListenersT, PayloadT>
```
Is a lot more descriptive compared to:
```
<T, X>
```